### PR TITLE
Get the oreg token from persistent credentials

### PR DESCRIPTION
**Pull request by pschiffe**
New secret variables are needed:
```
oreg_url: 'registry url for openshift'
oreg_api_url: 'https api url'
oreg_auth_user: 'username'
oreg_api_auth_password: 'password'
```

### CODE REVIEW

**Comment by pschiffe** [comment]
@kwoodson what about this?

File: playbooks/azure/openshift-cluster/build_gi.yml [hunk 1]
```diff
@@ -4,19 +4,26 @@
...
+  gather_facts: yes
   tasks:
   - name: "provision public ip address"
     import_role:
       name: openshift_azure
       tasks_from: provision_instance.yml
+  - name: get the oreg token
+    import_role:
+      name: openshift_azure
+      tasks_from: oreg_token.yml
```
Comment (kwoodson) [inline playbooks/azure/openshift-cluster/build_gi.yml]: I'm not sure this can be placed here as the `oc` binary is not yet on the node.

Comment (pschiffe) [reply playbooks/azure/openshift-cluster/build_gi.yml]: This runs from the host creating the infrastructure, so a workstation, or a container.

**Comment by kwoodson** [comment]
@pschiffe, after thinking about this.  I think we should move this to its own role.  This will probably be shared across AWS, Azure, etc.

What do you think?

**Comment by pschiffe** [comment]
@kwoodson possibly yes; I'll create a role.

**Comment by pschiffe** [comment]
@kwoodson I've moved this to a separate role. What do you think?

**Comment by kwoodson** [comment]
@pschiffe, Let's name the role `openshift_registry_token`.  Then ensure all variables have the role name as prefix: `openshift_registry_token_<var>`.  That's the pattern that's used in openshift-ansible.  After that then I'll tag with lgtm.

**Comment by kwoodson** [comment]
@pschiffe /lgtm

**Review by kwoodson — APPROVED** [review]

**Pull request merged**